### PR TITLE
Add calculation of rotation angle between 2 points (radians)

### PR DESCRIPTION
**Pull request by egarciamendez**
## Description

Add calculation of rotation angle between 2 points. 

15E41D45-93B5-4191-8FDF-AE46DE6EB535

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: blueprints/geometry/operations.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    RAD
+        rotation of end point relative to begin node in radians in the given plane [rad].
+
+    Raises
+    ------
+    ValueError
+        If start_point and end_point are the same.
+        If start_point or end_point do not have z value when rotation angle in XZ or YZ plane is requested.
+    """
+    if list(start_point.coords) == list(end_point.coords):
```
Comment (SZeltaat) [inline blueprints/geometry/operations.py:52]: Have you considered using the euqals_exact() method from Shapely?

https://shapely.readthedocs.io/en/stable/reference/shapely.Point.html#shapely.Point.equals_exact

File: blueprints/geometry/operations.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    if coordinate_system != CoordinateSystemOptions.XY and (not start_point.has_z or not end_point.has_z):
+        msg = f"Coordinate system '{coordinate_system}' requires z value in both points."
+        raise ValueError(msg)
+
+    # Map coordinate system to the corresponding attributes
+    coordinates_map = {
+        CoordinateSystemOptions.XY: ("x", "y"),
+        CoordinateSystemOptions.XZ: ("x", "z"),
+        CoordinateSystemOptions.YZ: ("y", "z"),
+    }
```
Comment (SZeltaat) [inline blueprints/geometry/operations.py:65]: Could this have been done in de CoordinateSystemOptions Enum?
Suggestion: You could use "xy" in the Enum and then use split here.

**Review by SZeltaat — APPROVED** [review]
Nice job Enrique!
I just have some suggestions and considerations! Good luck with that!

**Comment by egarciamendez** [comment]
@SZeltaat being sharp as always! Thank you very much for your review :-)

**Pull request merged**